### DISCUSSION TITLE
Celery briefly replaces %n in filenames with the queue names instead of the respective worker names on restart

### DISCUSSION THREAD

**Discussion by nrj1024** [celery/celery#8413] [Q&A]
# Checklist

- [x] I have checked the [issues list](https://github.com/celery/celery/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22Category%3A+Documentation%22+)
  for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [x] I have checked the [commit log](https://github.com/celery/celery/commits/main)
  to find out if the bug was already fixed in the main branch.
- [x] I have included all related issues and possible duplicate issues in this issue
      (If there are none, check this box anyway).

#### Related Issues

- None

#### Possible Duplicates

- None

# Description

I have noticed that whenever celery is restarted, the filenames for files such as PID file, LOG file where %n is there, is replaced with the list of queue names briefly instead of the actual worker name(s). This leads to a LockFailed error while reading the PID file in celery/platforms.py > write_pid | Line 211, due to file name being too long. But it gets fine once the worker names are there.

I tried to look it up but couldn't find anything relevant. Why is this happening? I am currently using version 4.2.2

# Suggestions

If this is supposed to be happening then it should be mentioned in the documentation for celery version 4.2.2 (or other versions too) under the section https://docs.celeryq.dev/en/stable/userguide/workers.html#variables-in-file-paths

**Comment by auvipy** [comment]
can you provide code sample to reproduce with with celery latest or in main branch?
